umh-core/pkg/service/s6/s6.go

```diff
@@ -1101,31 +1129,136 @@ func (s *DefaultService) GetS6ConfigFile(ctx context.Context, servicePath string
 	return content, nil
 }
 
-// ForceRemove removes a service from the S6 manager
-func (s *DefaultService) ForceRemove(ctx context.Context, servicePath string, fsService filesystem.Service) error {
+// ForceRemove tears down an S6 long-run service **unconditionally**.
+//
+// It is the “nuclear option” the manager calls after several failed
+// graceful removals.  The function must therefore:
+//
+//   - **Be idempotent & non-blocking** – safe to invoke every 100 ms;
+//     never waits or polls.
+//   - **Erase every artefact** the service could have left behind.
+//   - **Return nil only when nothing remains.**
+//
+// Removal sequence
+// ----------------
+//  1. Best-effort **stop** the daemon *and* its logger with `s6-svc -d`.
+//     This is fast and harmless even if they are already down.
+//  2. **SIGTERM any lingering `s6-supervise` processes** (main + logger)
+//     by reading `<servicedir>/supervise/pid`.  Without this step a still-
+//     running supervisor could re-create the `supervise/` directory after
+//     we delete the tree.
+//  3. **Delete the two artefact trees**
+//     <servicePath>                       (includes …/log subdir)
+//     <logsBase>/<serviceName>            (rotated log files)
+//  4. **Double-check** that both paths are truly gone and that no I/O
+//     errors occurred; otherwise return an error so the caller retries.
+func (s *DefaultService) ForceRemove(
+	ctx context.Context,
+	servicePath string,
+	fsService filesystem.Service,
+) error {
+	start := time.Now()
+	defer func() {
+		metrics.ObserveReconcileTime(
+			metrics.ComponentS6Service,
+			servicePath+".forceRemove",
+			time.Since(start))
+	}()
+
 	if ctx.Err() != nil {
 		return ctx.Err()
 	}
 
-	err := fsService.RemoveAll(ctx, servicePath)
-	if err != nil {
-		return fmt.Errorf("failed to remove service: %w", err)
+	//--------------------------------------------------------------------
+	// 1. Best-effort graceful stop (idempotent, non-blocking)
+	//--------------------------------------------------------------------
+	mainErr := s.Stop(ctx, servicePath, fsService)                         // main
+	loggerErr := s.Stop(ctx, filepath.Join(servicePath, "log"), fsService) // logger
+
+	//--------------------------------------------------------------------
+	// 2. SIGTERM lingering s6-supervise processes to avoid resurrection
+	//--------------------------------------------------------------------
+	killSupervise := func(dir string) error {
+		pidFile := filepath.Join(dir, "supervise", "pid")
+		data, err := fsService.ReadFile(ctx, pidFile)
+		if err != nil || len(data) == 0 {
+			return nil // no supervisor ⇒ nothing to kill
+		}
+		if pid, err := strconv.Atoi(strings.TrimSpace(string(data))); err == nil && pid > 0 {
+			// Best effort: ignore EPERM/ESRCH etc.
+			err = syscall.Kill(pid, syscall.SIGTERM)
```

Comment:
How does this ignore ESRCH or EPERM ?

Reply:
Comment needs to be removed there, correct